src/App.tsx

```diff
@@ -1,204 +1,84 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import { UserWarning } from './Components/UserWarning/UserWarning';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { ErrorMessage } from './Components/ErrorMessage';
+import { Todo } from './types/Todo';
+import { getTodos, getTodosByStatus } from './api/todos';
+import { Errors } from './types/Errors';
+import { Status } from './types/Status';
+import { Header } from './Components/Header';
+import { USER_ID } from './utils/userId';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
-
-  return (
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
```

Comment:
it is redundant to keep state for visible todos.
filter todos before render. whenever will change filterStatus - app will rerender and filter todos with current filter status.